domain/machine/service/provider.go

```diff
@@ -51,95 +54,60 @@ func NewProviderService(
 	}
 }
 
-// CreateMachine creates the specified machine. The nonce is an optional
-// parameter and is used only used during bootstrapping to ensure that
-// the machine is created with a unique name.
-func (s *ProviderService) CreateMachine(ctx context.Context, args CreateMachineArgs) (machine.UUID, machine.Name, error) {
+// AddMachine creates the net node and machines if required, depending
+// on the placement.
+// It returns the net node UUID for the machine and a list of child
+// machine names that were created as part of the placement.
+//
+// The following errors can be expected:
+// - [machineerrors.MachineNotFound] if the parent machine (for container
+// placement) does not exist.
+func (s *ProviderService) AddMachine(ctx context.Context, args domainmachine.AddMachineArgs) (string, []machine.Name, error) {
```

Comment:
1) `and a list of machine names that were created, starting with the parent followed by any child machines`
2) I don't think we should return the `nodeUUID`. If I remember correctly, we don't want the net node concept to leak outside of the domain. The fact we don't actually use it anywhere attests to this (what would we use it for?

Reply:
In fact, instead of returning a slice of machine names, maybe consider returning an instance of:
```
// think of a better name if you can!
type AddMachineResults strict {
    Machine       machine.Name
    ChildMachines []machine.name
}
```

Reply:
It would only ever be:

```go
type AddMachineResults strict {
    Machine       machine.Name
    ChildMachine *machine.name
}
```

Reply:
Done, see 90448c6a79afa5ed66896fcfb2ad8e274a4c942b